### PR TITLE
SONARPY-2001:Create GeneratedIPythonFile to hold offset data

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

**Review by ghislainpiot — CHANGES_REQUESTED** [review]
I think we only need to store one map in the GeneratedIPythonFile.

I also don't understand how it works, shouldn't there be a way to overwrite the contents of the file ?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/GeneratedIPythonFile.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+public class GeneratedIPythonFile implements PythonInputFile {
+
+  InputFile wrappedFile;
+
+  Map<Integer, Offset> offsetMap;
+  // A map from the line to a Map of column location of escaped chars: 
+  // from the position in the Python code to the position in the original file
+  Map<Integer, Map<Integer, Integer>> colOffSet;
+
+  public GeneratedIPythonFile(InputFile wrappedFile, Map<Integer, Offset> offsetMap, Map<Integer, Map<Integer, Integer>> colOffSet) {
```
[the remark this answers is not in the document]
Comment (joke1196) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/GeneratedIPythonFile.java:34]: You are right we can only rely on the IPython location but we need more than just one column offset per line. And yes sorry I forgot the aggregated content.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/GeneratedIPythonFile.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+public class GeneratedIPythonFile implements PythonInputFile {
+
+  InputFile wrappedFile;
+
+  Map<Integer, IPythonLocation> offsetMap;
+
+  private String pythonContent;
+
+  public GeneratedIPythonFile(InputFile wrappedFile, String pythonContent, Map<Integer, IPythonLocation> offsetMap) {
```
Comment (ghislainpiot) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/GeneratedIPythonFile.java:35]: I think adding the location map and offset map should be done in the ticket that needs it, that way we avoid going back and forth for something that can still change.

**Review by ghislainpiot — APPROVED** [review]
LGTM!

**Pull request merged**